app/src/main/java/com/infomaniak/mail/utils/extensions/AttachmentExt.kt

```diff
@@ -129,42 +123,32 @@ object AttachmentExt {
     //endregion
 
     suspend fun Attachment.startUpload(draftLocalUuid: String, mailbox: Mailbox, realm: Realm) {
-        val attachmentFile = getUploadLocalFile().also {
-            if (it?.exists() != true) {
+        val attachmentFile = getUploadLocalFile().let { file ->
+            if (file?.exists() != true) {
                 SentryLog.d(ATTACHMENT_TAG, "No local file for attachment $name")
                 throw UploadMissingLocalFileException()
             }
+            file
         }
 
         val userApiToken = AccountUtils.getUserById(mailbox.userId)?.apiToken?.accessToken ?: return
-        val headers = HttpUtils.getHeaders(contentType = null).newBuilder()
-            .set("Authorization", "Bearer $userApiToken")
-            .addUnsafeNonAscii("x-ws-attachment-filename", name)
-            .add("x-ws-attachment-mime-type", mimeType)
-            .add("x-ws-attachment-disposition", "attachment")
-            .build()
-        val request = Request.Builder().url(ApiRoutes.createAttachment(mailbox.uuid))
-            .headers(headers)
-            .post(attachmentFile!!.asRequestBody(mimeType.toMediaType()))
-            .build()
-
-        val response = AccountUtils.getHttpClient(mailbox.userId).newCall(request).execute()
-
-        val apiResponse = ApiController.json.decodeFromString<ApiResponse<Attachment>>(response.body?.string() ?: "")
-        if (apiResponse.isSuccess() && apiResponse.data != null) {
+        val okHttpClient = AccountUtils.getHttpClient(mailbox.userId)
+        val apiResponse = ApiRepository.createAttachments(attachmentFile, mailbox, name, mimeType, userApiToken, okHttpClient)
+
+        if (apiResponse != null && apiResponse.isSuccess() && apiResponse.data != null) {
```

Comment:
apiResponse?.data != true should be enough
And either way you can merge the first two clauses like this : `apiResponse?.isSuccess() == true`